visma/matrix/structure.py

```diff
@@ -1,4 +1,7 @@
 from visma.functions.constant import Constant
+from visma.functions.structure import Expression
+from visma.simplify.simplify import simplify
```

Comment:
Hey I looked into your code and after building it, I found that the problem was with this import which was causing a Circular import problem.
You can fix this by removing this line from here and adding it later just at the start of your function.